cli.py

```diff
@@ -0,0 +1,187 @@
+        return folder_flags_parsing(args)
+
+    logger.error("This should be unreachable. If you see this message, most likely a new extension was not implemented correctly.")
+    return []
+
+
+def file_flags_parsing(args) -> List[Tuple[str, str, Optional[str]]]:
+    input_file, output_file, correct_file = args.input_file, args.output_file, args.correct_file
+    if input_file is None:
+        logger.error("Specify the input file")
+        sys.exit(1)
+    if not output_file:
+        output_file = utils.filename_remove_extension(input_file)+".json"
+    return [(input_file, output_file, correct_file)]
+
+
+def folder_flags_parsing(args) -> List[Tuple[str, str, Optional[str]]]:
+    # todo: check for trailing slash
+    input_folder, output_folder, correct_folder = args.input_folder, args.output_folder, args.correct_folder
+    if input_folder is None:
+        logger.error("Specify the input folder")
+        sys.exit(1)
+    if not output_folder:
+        output_folder = input_folder
+
+    files: List[str] = []
+    try:
+        files = utils.list_input_files(input_folder)
+    except Exception as e:
+        logger.error("Todo: check how can i invoke this exception")
+        sys.exit(1)
+
+    answer: List[Tuple[str, str, Optional[str]]] = []
+
+    for single_filename in files:
+        filename_segment = utils.filename_from_path(single_filename)
+
+        output_filename_segment = utils.filename_remove_extension(filename_segment)
+        if input_folder == output_folder:
+            output_filename_segment += ".out"
+        output_filename_segment += ".json"
+        output_filename = os.path.join(output_folder, output_filename_segment)
+
+        correct_filename = None
+        if correct_folder:
+            correct_filename_segment = utils.filename_remove_extension(filename_segment) + ".json"
+            correct_filename = os.path.join(correct_folder, correct_filename_segment)
+
+        answer.append((single_filename, output_filename, correct_filename))
+
+    return answer
+
+
+def check_folders_existance(filename_tuples: List[Tuple[str, str, Optional[str]]], create_non_existent_folders=False):
+    unique_folder = set()
+    for x in filename_tuples:
+        output_filename_path = x[1]
+        output_folder_path = os.path.dirname(output_filename_path)
+        unique_folder.add(output_folder_path)
+
+    for x in unique_folder:
+        if not os.path.isdir(x):
+            logger.error(f"One of the output folders, doesn't exist: {x}")
+            if create_non_existent_folders:
+                logger.debug("Trying to create the output folder")
+                utils.mkdir(x)
+
+
+def evaluate_files(filename_tuples: List[Tuple[str, str, Optional[str]]]) -> None:
+    for single_file_tuple in filename_tuples:
+        logger.debug(single_file_tuple)
+        input_filename, output_filename, correct_filename = single_file_tuple
+
+        try:
+            lines = utils.load_file(input_filename)

```

Comment:
This line crashed when the file contained invalid character (for example file with only 0xffffffff). I changed it to handle UnicodeDecodeError.